contracts/duckies/nft/DucklingFamilyGame/DucklingFamilyGameV1.sol

```diff
@@ -0,0 +1,245 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.18;
+
+import '@openzeppelin/contracts-upgradeable/utils/CountersUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/utils/StringsUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/utils/cryptography/ECDSAUpgradeable.sol';
+
+import '../../../interfaces/IVoucher.sol';
+import '../RandomUpgradeable.sol';
+import '../Gene.sol';
+
+contract DucklingFamily is IVoucher, RandomUpgradeable {
+	using CountersUpgradeable for CountersUpgradeable.Counter;
+	using Gene for uint256;
+
+	// ------- IVoucher -------
+
+	enum VouhcherActions {
+		MintPacks,
+		MeldFlocks
+	}
+
+	function useVouchers(Voucher[] calldata vouchers, bytes calldata signature) external {
+		// use vouchers to mint/meld NFTs
+	}
+
+	function useVoucher(Voucher calldata voucher, bytes calldata signature) external {
+		// use voucher to mint/meld NFTs
+	}
+
+	// ------- Ducklings Game -------
+
+	uint8 public constant MAX_PACK_SIZE = 100;
+	uint8 public constant FLOCK_SIZE = 5;
+
+	enum Collections {
+		Ducklings,
+		Zombeaks,
+		SuperLegendary
+	}
+
+	enum Classes {
+		Common,
+		Rare,
+		Epic,
+		Legendary
+	}
+
+	uint8[] internal rarityWeights; // 70, 20, 5, 1
+	uint8[] internal meldingZombeakWeights; // 10, 5, 2, 0
```

Comment:
`Fix array to uint8 [5]`

Suggesting:
probabilityOfRarity
probabilityOfMutation
or

chanceOfRarity
changeofMutation

Reply:
It is better to declare dynamic arrays, as Solidity still can't convert between dynamic and fixed, and vice versa :<(

Still, renamed: f1e7948f00ec9ab95587b20e61774e67a8048737